velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -60,8 +60,63 @@ core::SortOrder toSortOrder(const ::substrait::SortField& sortField) {
   }
 }
 
+/// Holds the information required to create
+/// a project node to simulate the emit
+/// behavior in Substrait.
+struct EmitInfo {
+  std::vector<core::TypedExprPtr> expressions;
+  std::vector<std::string> projectNames;
+};
+
+/// Helper function to extract the attributes required to create a ProjectNode
+/// used for interpretting Substrait Emit.
+EmitInfo getEmitInfo(
+    const ::substrait::RelCommon& relCommon,
+    const core::PlanNodePtr& node) {
+  const auto& emit = relCommon.emit();
+  int emitSize = emit.output_mapping_size();
+  EmitInfo emitInfo;
+  emitInfo.projectNames.reserve(emitSize);
+  emitInfo.expressions.reserve(emitSize);
+  const auto& outputType = node->outputType();
+  for (int i = 0; i < emitSize; i++) {
+    int32_t mapId = emit.output_mapping(i);
+    emitInfo.projectNames[i] = outputType->nameOf(mapId);
+    emitInfo.expressions[i] = std::make_shared<core::FieldAccessTypedExpr>(
+        outputType->childAt(mapId), outputType->nameOf(mapId));
+  }
+  return emitInfo;
+}
+
 } // namespace
 
+core::PlanNodePtr SubstraitVeloxPlanConverter::processEmit(
+    const ::substrait::RelCommon& relCommon,
+    const core::PlanNodePtr& noEmitNode) {
+  switch (relCommon.emit_kind_case()) {
+    case ::substrait::RelCommon::EmitKindCase::kDirect:
+      return noEmitNode;
+    case ::substrait::RelCommon::EmitKindCase::kEmit: {
+      auto emitInfo = getEmitInfo(relCommon, noEmitNode);
+      return std::make_shared<core::ProjectNode>(
+          nextPlanNodeId(),
+          std::move(emitInfo.projectNames),
+          std::move(emitInfo.expressions),
+          std::move(noEmitNode));
+    }
+    default:
+      VELOX_FAIL("unrecognized emit kind");
+  }
+}
+
+/// RelCommon related validations evaluated
+void ValidateEmitExclusion(const ::substrait::RelCommon& relCommon) {
+  if (relCommon.has_emit()) {
+    VELOX_FAIL(
```

Comment:
Use VELOX_USER_CHECK(!relCommon.has_emit(), "...");

Make sure method and function names start with lower case: ValidateEmitExclusion -> validateEmitExclusion